File: max_sum_k_consecutive_num.cpp
#include<iostream>

using namespace std;

int maxSum(int* arr, int n, int k)
{
    if (n<k)
    {
        cout << "Invalid Operation";
        return -1;
    }

    // finding out sum of first window 
    int win_sum = 0;  
    for(int i=0; i<k; i++)
        win_sum = win_sum + arr[i];
    
    int max_sum = win_sum;
    // printf("%d", win_sum);

    // finding sum of remaining windows by removing element from prev window and 
    // adding next elem to current window
    for (int i=0; i<n-k; i++)
    {
        win_sum = win_sum - arr[i] + arr[i+k];
        max_sum = max(win_sum, max_sum); 
    }

    // can also be done as follows
    // for (int i=k; i<n; i++)
    // {
    //     win_sum += arr[i] - arr[i-k];
    //     max_sum = max(win_sum, max_sum);
    // }

    return max_sum;
    
}


int main()
{
    int arr[] = {80, -50, 90, 100};
    int n = sizeof(arr)/sizeof(arr[0]);
    int k = 2; //window size
    cout << maxSum(arr, n, k);
    return 0;
}

File: max_sum_k_consecutive_num.py
def maxSum(arr, k):
    n = len (arr)

    if (n <= k): # array size is less than window size
        print("Invalid Operation")
        return -1 

    win_sum = sum (arr[i] for i in range(k))
    max_sum = win_sum
    # print(max_sum)

    for i in range (n-k):
        win_sum = win_sum - arr[i] + arr[i+k]
        max_sum = max(max_sum, win_sum)
        
    return max_sum
    



arr = [80, -50, 90, 100]
k = 2 # k is window size,  so ans should be 190
result = maxSum(arr, k)
print(f"maxisum sum of {k} consecutive numbers is {result}")

